apps/pagetour-sdk/src/playtour/pagetourplay.ts

```diff
@@ -1122,7 +1172,61 @@ class PageTourPlay {
         stepHeadingElement.innerText = this.tour.steps[stepCount].headerText
         stepDescriptionElement.innerHTML = stepDescription
 
-        //this.tether = this.getTetherObject(stepCount)
+        let annoFeedbackOpts = opts.feedback.AnnouncementFeedbackOptions;
+        let defaultAnnoFeedbackOpts = this.configStore.DefaultOptions.feedback.AnnouncementFeedbackOptions;
+        
+        if(annoFeedbackOpts.enabled && action != RunTourAction.Preview){
+          let annoFeedbackType = annoFeedbackOpts.type === undefined ? defaultAnnoFeedbackOpts.type : annoFeedbackOpts.type;
+          var ratingElement: NodeListOf<HTMLInputElement>;
+          if(this.announcementFeedbackObj[this.currentStep].submitted == false){
+            this.updateAnnouncementFeedbackElement();
+            if(annoFeedbackType == 'like-dislike'){
+              let likeratingRadioButtons = document.querySelector('input[name="likerating"]:checked') as HTMLInputElement;
+              if(likeratingRadioButtons) likeratingRadioButtons.checked = false;
+              ratingElement = document.getElementsByName('likerating') as NodeListOf<HTMLInputElement>;
+            }
+            else if(annoFeedbackType == 'yes-no'){
+              let yesratingRadio = document.querySelector('input[name="yesnorating"]:checked') as HTMLInputElement;
+              if(yesratingRadio) yesratingRadio.checked = false;
+              ratingElement = document.getElementsByName('yesnorating') as NodeListOf<HTMLInputElement>;
+            }
+
+            var self = this;  
+            var rating =0.0;
+            
+            for(var i=0; i<ratingElement.length; i++){
+              ratingElement[i].addEventListener('click', function(e){     
+                   
+                if(this.checked &&  self.announcementFeedbackObj[self.currentStep].submitted==false){
+                  rating = parseFloat(this.value)/5;
```

Comment:
can we get this from config

Reply:
getting the rating value from the element directly without computation